drivers/serial/uart_nrfx_uart.c

```diff
@@ -878,10 +878,11 @@ static int uart_nrfx_irq_tx_ready_complete(const struct device *dev)
 	 * called after the TX interrupt is requested to be disabled but before
 	 * the disabling is actually performed (in the IRQ handler).
 	 */
-	return nrf_uart_int_enable_check(uart0_addr,
-					 NRF_UART_INT_MASK_TXDRDY) &&
-	       !disable_tx_irq &&
-	       event_txdrdy_check();
+	bool ready = nrf_uart_int_enable_check(uart0_addr,
+					       NRF_UART_INT_MASK_TXDRDY) &&
+		     !disable_tx_irq &&
+		     event_txdrdy_check();
+	return ready ? INT_MAX : 0;
```

Comment:
```suggestion
	return ready ? 1 : 0;
```
Only one byte is guaranteed to be accepted by `uart_nrfx_fifo_fill()`. Another one can be accepted only if the TXDRDY event gets generated again and that happens after the previous one is sent.
Additionally, `uart_nrfx_irq_tx_ready_complete()` is also used as `uart_tx_complete` and in this case it should return 1 when transmission is completed.

Reply:
You're right, I misread the `event_txdrdy_check()` condition in the `uart_nrfx_fifo_fill` loop.
Updated